sources/packages/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts

```diff
@@ -112,27 +119,45 @@ export class InstitutionLocationInstitutionsController extends BaseController {
    * Get all active application of a location in an institution
    * with application_status is completed.
    * @param locationId location id.
+   * @param pagination options to execute the pagination.
+   * @param archived archive value of applications requested by user.
    * @returns Student active application list of an institution location.
    */
   @HasLocationAccess("locationId")
   @Get(":locationId/active-applications")
   async getActiveApplications(
     @Param("locationId") locationId: number,
-  ): Promise<ActiveApplicationSummaryAPIOutDTO[]> {
+    @Query() pagination: ApplicationStatusPaginationOptionsAPIInDTO,
+    @Query("archived", ParseBoolPipe)
+    archived: boolean,
+  ): Promise<PaginatedResultsAPIOutDTO<ActiveApplicationSummaryAPIOutDTO>> {
     const applications = await this.applicationService.getActiveApplications(
       locationId,
+      pagination,
+      archived,
     );
-    return applications.map((eachApplication) => {
-      const offering = eachApplication.currentAssessment?.offering;
-      return {
-        applicationNumber: eachApplication.applicationNumber,
-        applicationId: eachApplication.id,
-        studyStartPeriod: getISODateOnlyString(offering?.studyStartDate),
-        studyEndPeriod: getISODateOnlyString(offering?.studyEndDate),
-        applicationStatus: eachApplication.applicationStatus,
-        fullName: getUserFullName(eachApplication.student.user),
-      };
-    });
+
+    return {
+      results: applications.results.map((eachApplication) => {
+        const offering = eachApplication.currentAssessment?.offering;
+        return {
+          applicationNumber: eachApplication.applicationNumber,
+          applicationId: eachApplication.id,
+          studyStartPeriod: getISODateOnlyString(offering?.studyStartDate),
+          studyEndPeriod: getISODateOnlyString(offering?.studyEndDate),
+          applicationStatus: eachApplication.applicationStatus,
+          applicationSholasticStandingStatus:
+            this.applicationService.getApplicationSholasticStandingStatus(
+              eachApplication.isArchived,
+              eachApplication.currentAssessment?.studentScholasticStanding?.id,
+            ),
+          fullName: getUserFullName(eachApplication.student.user),
+          scholasticStandingId:
+            eachApplication.currentAssessment?.studentScholasticStanding?.id,
+        };
+      }) as ActiveApplicationSummaryAPIOutDTO[],
```

Comment:
we can avoid as(as ActiveApplicationSummaryAPIOutDTO[]) here unless there is a need to use it. It will not highlight if any mandatory fields are missing.